### DISCUSSION TITLE
Workspace usage and overall feel

### DISCUSSION THREAD

**Discussion by andre-romano** [UCLA-IRL/ndn-workspace-solid#115] [General]
I would like to begin by congratulating the Workspace team for their tool, which demonstrates excellent functionality and user experience overall. However, I would like to highlight some points for potential improvement:

- **File Compilation Issues**: Workspace does not compile files without a .tex extension. It is recommended that Workspace include a logging console to inform users of the cause of such errors, facilitating easier troubleshooting. Although this feature is present in the selector next to the COMPILE button (under the “Log” option), its placement was not immediately evident. This indicates that the current interface may not be user-friendly. Consider positioning the log console similarly to Overleaf or at the bottom of the page, akin to embedded terminals in Visual Studio Code, PyCharm, or Eclipse UI designs.
  
- **Editing Restrictions Post-Compilation**: Upon compiling a PDF, the user is unable to edit the files without switching menus or tabs and then returning to the Editor. The display selector beside the COMPILE button does not function as expected. The “Both” option should ideally display the Editor and PDF side by side rather than one above the other (refer to the screenshot below). Similar issues were observed with the “PDF” and “Editor” options, where the PDF consistently overlays other HTML components.
  
- **File Synchronization Delays**: The file synchronization process typically takes between 2-4 minutes. This duration varied, being sometimes faster and other times slower. We attempted to mitigate potential interferences by restarting the browser application, using different browsers (current versions of Edge and Chrome), and utilizing Incognito mode to avoid caching and other browser-related interferences. It is suspected that these delays may stem from routing or sync agent convergence times. This synchronization time could significantly impede the widespread adoption of Workspace and warrants further investigation.
  
- **Concurrent File Modifications**: Another concern is the handling of simultaneous modifications to the same section of a file. Overleaf, the most widely adopted LaTeX online editor, occasionally encounters similar issues, necessitating a browser tab refresh (F5) to restore normal operation. If Workspace’s synchronization time is indeed in the order of minutes, this could pose a considerable challenge.
  
- **Multiple Browser Sessions**: We observed that when a user attempts to connect to Workspace using their credentials on different browsers (e.g., Edge and Chrome) or machines, the previous session is disconnected from the NDN testbed. While this outcome is likely intended, it is noteworthy.

These points are presented solely with the intention of aiding in the continuous improvement of the Workspace tool.

**Comment by zjkmxy** [comment]
Thank you for your comments.
- File Compilation Issues: I agree the UI needs to be changed.
- File Synchronization Delays: Could you provide more information on this? I know that our performance is slow but I don't think 2-4 minutes is common. How many edits are you done to the workspace?
- Concurrent File Modifications: This is not an issue. One of our design goal is to automatically resolve all conflicts.
- Multiple Browser Sessions: In the current design this is strictly forbidden. One identity is limited to be used in one tab, on one device.
